### ISSUE TITLE
{{#each a in b}} stops reactive updates from outside

### ISSUE THREAD

**Issue by wh0** [meteor/meteor#5181]
In Meteor 1.2-rc.15, the `each..in` template tag doesn't allow reactive references to values outside the tag.

```handlebars
{{#with getOrder}}
  <p>Give me {{qty}} of everything:</p>
  <ul>
    {{#each item in items}}
      <li>{{qty}}x {{item}}</li>
    {{/each}}
  </ul>
{{/with}}
```
(suppose `getOrder` is a reactive helper that returns something like `{qty: 5, items: ['pancakes']}`)

When `qty` changes, it'll update partially, to show:

> Give me 6 of overything:
> - 5x pancakes

test app: https://github.com/wh0/meteor-bug5181

**Comment by wh0** [comment]
cc @mitar

**Comment by wh0** [comment]
https://github.com/meteor/meteor/blob/81b2d3e6eecb5cedd9a14696824300acfdf3536c/packages/blaze/builtins.js#L178 basically snapshots the immediately surrounding context when each item view is created

**Comment by wh0** [comment]
added pr #5182

**Comment by stubailo** [comment]
Fixed, should come out in the next RC (17)

**Comment by stubailo** [comment]
Released now as part of `1.2-rc.17`!
